.repo/sample-list-generator/src/libs/api-detector.ts

```diff
@@ -0,0 +1,201 @@
+import {
+  ApiItem,
+  ApiItemWithType,
+  ApiTypeResult,
+  ExtensionApiMap
+} from '../types';
+import * as babel from '@babel/core';
+import { isIdentifier } from '@babel/types';
+import fs from 'fs/promises';
+import { getAllFiles } from '../utils/filesystem';
+import { loadExtensionApis } from './api-loader';
+
+let EXTENSION_API_MAP: ExtensionApiMap = loadExtensionApis();
+
+/**
+ * Gets the type of an api call.
+ * @param namespace - The namespace of the api call.
+ * @param propertyName - The property name of the api call.
+ * @returns The type of the api call.
+ * @example
+ * getApiType('tabs', 'query')
+ * // returns 'method'
+ */
+export const getApiType = (
+  namespace: string,
+  propertyName: string
+): ApiTypeResult => {
+  namespace = namespace.replace(/_/g, '.');
+
+  if (EXTENSION_API_MAP[namespace]) {
```

Comment:
nit: extract `EXTENSION_API_MAP[namespace]` to a variable

Reply:
Done. https://github.com/GoogleChrome/chrome-extensions-samples/pull/963/commits/f937230468614ee77c74d57d6174e7460fe2941b